### ISSUE TITLE
I want to know if anyone has correctly trained the first stage？

### ISSUE THREAD

**Issue by tdlhyj** [guoqincode/AnimateAnyone-unofficial#56]
The first stage, no matter how many training steps are taken, cannot achieve correct pose control

**Comment by tgxs002** [comment]
Same issue. You can try replacing "noisy_latents = noisy_latents + latents_pose" by "noisy_latents = torch.cat([noisy_latents, latents_pose], dim=1)", (also related codes accordingly) and then it should work.
 

https://github.com/guoqincode/AnimateAnyone-unofficial/assets/38932123/e2e30546-a923-4b2b-8e46-150997c1ba70

**Comment by tdlhyj** [comment]
> Same issue. You can try replacing "noisy_latents = noisy_latents + latents_pose" by "noisy_latents = torch.cat([noisy_latents, latents_pose], dim=1)", (also related codes accordingly) and then it should work.
> 
>  grid.mp4

Okay, I'll give it a try. how many steps did you train for this

**Comment by tgxs002** [comment]
2000 iterations for bs 40, lr 5e-6, resuming from a converged checkpoint with "plus condition"

**Comment by guoqincode** [comment]
> Same issue. You can try replacing "noisy_latents = noisy_latents + latents_pose" by "noisy_latents = torch.cat([noisy_latents, latents_pose], dim=1)", (also related codes accordingly) and then it should work.
> 
>  grid.mp4

Great! I tried this scheme before with Stage 1 training, but I didn't train with stage2.

**Comment by lbwang2006** [comment]
or change input channel of unet = 8？

**Comment by tgxs002** [comment]
correctly, I appended 4 channels of zeros to "conv_in.weight", and change the input channel to 8. similar to the implementation of runwayml/stable-diffusion-inpainting.

**Comment by lbwang2006** [comment]
> correctly, I appended 4 channels of zeros to "conv_in.weight", and change the input channel to 8. similar to the implementation of runwayml/stable-diffusion-inpainting.

both unet and referencenet is modified?

**Comment by lbwang2006** [comment]
and why changed to plus condition again?

> correctly, I appended 4 channels of zeros to "conv_in.weight", and change the input channel to 8. similar to the implementation of runwayml/stable-diffusion-inpainting.

**Comment by tgxs002** [comment]
> > correctly, I appended 4 channels of zeros to "conv_in.weight", and change the input channel to 8. similar to the implementation of runwayml/stable-diffusion-inpainting.
> 
> both unet and referencenet is modified?

The referencenet works pretty well, so I didn't touch it.

**Comment by jaymefosa** [comment]
@tgxs002 did you also do a 10% dropout during training, with an inference CFG like a cat([zeros_like(pose_latents), pose_latents]) ?

**Comment by tgxs002** [comment]
> @tgxs002 did you also do a 10% dropout during training, with an inference CFG like a cat([zeros_like(pose_latents), pose_latents]) ?

I didn't train with 10% dropout, and it doesn't seem critical for inference.

**Comment by ireneMsm2020** [comment]
@tgxs002  did you trained the "conv_in.weight" ?

**Comment by tczuo** [comment]
> > @tgxs002 did you also do a 10% dropout during training, with an inference CFG like a cat([zeros_like(pose_latents), pose_latents]) ?
> 
> I didn't train with 10% dropout, and it doesn't seem critical for inference.

Could you share you stage1 weights? Thanks. I concat(noise,pose_latent) but the texture is not good。

**Comment by tgxs002** [comment]
> > > @tgxs002 did you also do a 10% dropout during training, with an inference CFG like a cat([zeros_like(pose_latents), pose_latents]) ?
> > 
> > 
> > I didn't train with 10% dropout, and it doesn't seem critical for inference.
> 
> Could you share you stage1 weights? Thanks. I concat(noise,pose_latent) but the texture is not good。 @tgxs002

In my experiments, the released code works pretty well for generating texture. You may find it helpful to check your input resolution.

**Comment by lbwang2006** [comment]
> > > > @tgxs002 did you also do a 10% dropout during training, with an inference CFG like a cat([zeros_like(pose_latents), pose_latents]) ?
> > > 
> > > 
> > > I didn't train with 10% dropout, and it doesn't seem critical for inference.
> > 
> > 
> > Could you share you stage1 weights? Thanks. I concat(noise,pose_latent) but the texture is not good。 @tgxs002
> 
> In my experiments, the released code works pretty well for generating texture. You may find it helpful to check your input resolution.

what about your stage 2 results ， the results of UBC is bad background in my experiment

**Comment by tgxs002** [comment]
> 
> what about your stage 2 results ， the results of UBC is bad background in my experiment
Yes, I also noticed the issue. Overfitting stage 1 helps, although it is not a scalable solution.

**Comment by tczuo** [comment]
> > > > @tgxs002 did you also do a 10% dropout during training, with an inference CFG like a cat([zeros_like(pose_latents), pose_latents]) ?
> > > 
> > > 
> > > I didn't train with 10% dropout, and it doesn't seem critical for inference.
> > 
> > 
> > Could you share you stage1 weights? Thanks. I concat(noise,pose_latent) but the texture is not good。 @tgxs002
> 
> In my experiments, the released code works pretty well for generating texture. You may find it helpful to check your input resolution.

Great!  my input resolution set as 512x768, work no way. But set as 512x512, it work very quickly.
![image](https://github.com/guoqincode/AnimateAnyone-unofficial/assets/9945763/bd6fde67-6d3e-4729-9898-d7e234172cd8)

**Comment by garychan22** [comment]
have you guys encountered the issue that utilizing the clip feature makes the color in the generated image weird like the patch in the cloth, after removing the clip feature the result becomes normal...

![image](https://github.com/guoqincode/Open-AnimateAnyone/assets/108175311/1ea151e5-cb65-47f3-8a49-3417b1fb1c02)

**Comment by newcherryberry** [comment]
> > > > > @tgxs002 did you also do a 10% dropout during training, with an inference CFG like a cat([zeros_like(pose_latents), pose_latents]) ?
> > > > 
> > > > 
> > > > I didn't train with 10% dropout, and it doesn't seem critical for inference.
> > > 
> > > 
> > > Could you share you stage1 weights? Thanks. I concat(noise,pose_latent) but the texture is not good。 @tgxs002
> > 
> > 
> > In my experiments, the released code works pretty well for generating texture. You may find it helpful to check your input resolution.
> 
> what about your stage 2 results ， the results of UBC is bad background in my experiment

@lbwang2006 Hi Lb, I wanted to discuss this issue, but couldn't find a way to contact you. Please let me know if I could potentially have your email or you could maybe email me.
